macros/src/lib.rs

```diff
@@ -63,12 +63,15 @@ pub fn derive_dynamic_bundle_clone(input: TokenStream) -> TokenStream {
     .into()
 }
 
-/// Implement `Query` for a struct
+/// Implement `Query` for a struct or enum.
 ///
-/// Queries structs can be passed to the type parameter of `World::query`. They must have exactly
+/// Queries can be passed to the type parameter of `World::query`. They must have exactly
 /// one lifetime parameter, and all of their fields must be queries (e.g. references) using that
 /// lifetime.
 ///
+/// For enum queries, the result will always be the first variant that satisfies the entity.
+/// Unit variants and variants without any fields will always satisfy an entity.
```

Comment:
An entity satisfies a query, not the reverse.

Reply:
Does "match" work instead? [c31467e](https://github.com/Ralith/hecs/pull/384/commits/c31467e7b93fdfc07818812ada266eb510b2e48d)

Reply:
Sure.